Refactor/login screen

itsahmedsalah opened the pull request.

itsahmedsalah committed:
Add new vector icons for login screen and update layout styles

itsahmedsalah committed:
Replace WebP logo with PNG version

- Delete `cinevers_logo.webp` from all drawable density folders.
- Add `cinevers_logo.png` to all drawable density folders.

itsahmedsalah committed:
Refactor login input field and update user icon

- Add `text_input_bg.xml` drawable for styled text input background.
- Update `fragment_login.xml`:
    - Modify `TextInputLayout` to remove default box background and hint animation.
    - Change start icon from `ic_profile` to `user`.
    - Apply `text_input_bg` as background to `TextInputEditText`.
    - Adjust `TextInputEditText` gravity, text color, and hint text color.
- Update `user.xml` to use `@color/shade_tertiary` for fill color.

itsahmedsalah committed:
Refactor login screen layout and update string resources

itsahmedsalah committed:
Add button background selector and update login screen layout with new buttons

itsahmedsalah committed:
Add button background drawable and update login screen button styles

itsahmedsalah committed:
Merge remote-tracking branch 'origin/develop' into refactor/login-screen

# Conflicts:
#	app/src/main/res/values/strings.xml

baraa0abd reviewed (changes requested):
Please add the descriptions and images of the changes that have been made.

baraa0abd committed:
Merge branch 'develop' into refactor/login-screen

MuhammedWael9991 commented:
Fix these issues and resolve the conflict

<img width="352" height="739" alt="image" src="https://github.com/user-attachments/assets/b9b75f45-b91c-44d1-991d-140133bb4786" />

MuhammedWael9991 reviewed (changes requested):
Fix the issue, please

itsahmedsalah committed:
Refactor login input fields and update text input background

- Update `text_input_bg.xml`:
    - Change root element from `<shape>` to `<selector>`.
    - Add different states (activated, focused, default) with specific stroke colors and consistent corner radius.
- Update `fragment_login.xml`:
    - Replace `TextInputLayout` and `TextInputEditText` for username and password with `EditText`.
    - Set fixed height for input fields.
    - Add `drawableStart` and `drawablePadding` to `EditText` for icons.
    - Adjust `drawableEnd` for password field.
    - Update constraints for `tv_forget_password`.

itsahmedsalah commented:
<img width="391" height="723" alt="image" src="https://github.com/user-attachments/assets/828d05c2-9a96-4c93-960c-897b3c729f53" />
Fixed the padding and hint text problems and also add the states of the edit text (needed to change the text input type to edit text)

itsahmedsalah committed:
Merge remote-tracking branch 'origin/develop' into refactor/login-screen

# Conflicts:
#	app/src/main/res/values/strings.xml

MuhammedWael9991 committed:
Merge remote-tracking branch 'origin/develop' into refactor/login-screen

itsahmedsalah committed:
Refactor login screen button layout for improved alignment and accessibility

itsahmedsalah committed:
Merge remote-tracking branch 'origin/refactor/login-screen' into refactor/login-screen

itsahmedsalah committed:
Merge remote-tracking branch 'origin/develop' into refactor/login-screen

NadeenXo commented on app/src/main/res/layout/fragment_login.xml:
```diff
@@ -15,94 +15,163 @@
+            style="@style/Regular.Medium"
+            android:layout_width="match_parent"
+            android:layout_height="wrap_content"
             android:layout_marginHorizontal="16dp"
```
use size from dimens file
example here ` android:layout_marginBottom="@dimen/spacing_medium"`

NadeenXo replied:
and the same for other spacing in the code

NadeenXo commented on app/src/main/res/layout/fragment_login.xml:
```diff
@@ -15,94 +15,163 @@
+
+        <TextView
+            android:id="@+id/tv_username"
+            style="@style/Regular.Medium"
```
I think it can be `  style="@style/Regular.Medium.Secondary"` 
So you can delete `  android:textColor="@color/shade_secondary_color"` 
Correct me if I'm wrong, please

YaseenOmar reviewed (changes requested):
When entering the username and password, the login button remains disabled.

YaseenOmar committed:
Refactor: Update Login screen UI and validation

- Update `fragment_login.xml`:
    - Replace `EditText` for username and password with `TextInputLayout` and `TextInputEditText` for improved UI and error handling.
    - Update styles, constraints, and IDs for various views.
    - Change progress indicator color and size.
    - Adjust button styles and properties.
- Update `text_input_bg.xml`:
    - Remove activated and focused states.
    - Update corner radius to `@dimen/radius_lg`.
- Update `MainActivity.kt`:
    - Add `bottomNavContainer` and manage its visibility based on the current fragment.
- Add `button_primary_selector.xml` for button background states.
- Update `item_tv_show_details_header.xml`, `item_tvshow_review.xml`, `item_movie_review.xml`, and `item_movie_detail_header.xml`:
    - Rename `imageView` to `image_view_logo`.
    - Rename `user_name` to `edit_text_user_name` in review item layouts.
- Update `colors.xml`:
    - Change `button_onSecondary` color.
- Update `ValidateFiledUseCase.kt`:
    - Modify username validation to allow only letters, numbers, and underscores.
- Update `dimens.xml`:
    - Add new spacing dimensions.
- Update `strings.xml`:
    - Add `logo` string resource.
- Add `button_login_text_selector.xml` for button text color states.
- Add `eye` drawable resource.

YaseenOmar committed:
Merge remote-tracking branch 'origin/develop' into refactor/login-screen

# Conflicts:
#	app/src/main/res/layout/item_movie_detail_header.xml
#	app/src/main/res/layout/item_tv_show_details_header.xml
#	app/src/main/res/values/dimens.xml
#	app/src/main/res/values/strings.xml

YaseenOmar committed:
Refactor: Adjust spacing and margins in item_season.xml
